test/test_redis_push.f90

```diff
@@ -1,26 +1,42 @@
-program name
-  use redis_mod
-  implicit none
-  integer, parameter ::n=200, m=100
-  integer :: dims(2)
-  real :: f(n,m)
-  integer ndims
-  integer i, j
+PROGRAM name
+  USE redis_mod
+  IMPLICIT NONE
+  INTEGER, PARAMETER ::n=200, m=100
+  INTEGER :: dims(2)
+  REAL*4 :: f4(n,m)
+  REAL*8 :: f8(n,m)
+  INTEGER*2 :: i2(n,m)
+  INTEGER*4 :: i4(n,m)
+  INTEGER ndims
+  INTEGER i, j
 
-  type (c_ptr) :: redis
+  TYPE (c_ptr) :: redis
 
   redis = setup_connection()
 
-  ! initialize array
-  do j=1,m
-    do i=1,n
-      f(i,j) = i + .5
-    end do
-  end do
+  ! initialize arrays
+  DO j=1,m
+     DO i=1,n
+        f4(i,j) = i + .5
+        f8(i,j) = i + .5
+        i2(i,j) = i + 1
+        i4(i,j) = i + 1
+     END DO
+  END DO
 
   ndims = 2
 
-  call redis_push(redis, 'A', f, 'f4', shape(f))
+  ! top level API
+  CALL redis_push(redis, 'A-f4', f4)
```

Comment:
These keys should each be tested in `test/test_send.py:test_redis_push`.

Reply:
done.